perfkitbenchmarker/scripts/step_load_driver/launch_mysql_service.py

```diff
@@ -0,0 +1,258 @@
+#!/usr/bin/env python
+
+# Copyright 2017 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""launch_mysql_service is a wrapper class for mysql_service_benchmark.
+It will call mysql_service_benchmark with varying thread counts depending on
+thread_count_list flag values.
+
+If a run_uri flag is provided, the program assumes that the PKB instance
+has been provisioned and prepared, and will only execute the run phase of PKB.
+If no run_uri is given, the program will execute the provision and prepare
+phase followed by consecutive run phases and then the cleanup and teardown
+phase. Below are examples of possible ways to call this class.
+
+Possible call:
+./launch_mysql_service --thread_count_list=[1,2,5] --sysbench_run_seconds=20
+./launch_mysql_service --run_uri=12e4s6t8 --thread_count_list=[1,2,4,8,10]
+
+All requirements for mysql_service_benchmark still apply. See
+perkfitbenchmarker.linuxbenchmarks.mysql_service_benchmark for more details.
+
+Any additional flags not specifically outlined in the flags below can be added
+as a list of strings under the 'additional_flags' flag. For example, a possible
+call with additional flags could be:
+./launch_mysql_service --run_uri=2u2u2u3i
+  --additional_flags=['--storage_size=100', '--cloud=GCP'].
+"""
+
+import datetime
+import logging
+import re
+import shlex
+import subprocess
+import sys
+import time
+import gflags
+
+# GLOBAL STRINGS
+SYSBENCH_RUN_SECONDS = 'sysbench_run_seconds'
+SYSBENCH_THREAD_COUNT = 'sysbench_thread_count'
+SYSBENCH_REPORT_INTERVAL = 'sysbench_report_interval'
+STORAGE_BUCKET = 'storage_bucket'
+THREAD_COUNT_LIST = 'thread_count_list'
+RUN_URI = 'run_uri'
+DEFAULT_RUN_TIME = 60
+STDOUT = 'STDOUT'
+STDERR = 'STDERR'
+DATETIME_FORMAT = '{:%m_%d_%Y_%H_%M_}'
+URI_REGEX = r'run_uri=([a-z0-9]{8})'
```

Comment:
looks like this string is used to check for run completion.

but I don't think run_uri=*** is a unique (enough) signal to indicate a particular stage is completed.

It's entirely possible that PKB code may emit some messages like "I am starting a run with run_uri=*** " at the beginning of a run.

I think we need to look for more specific strings to robustly catch the completion of a particular run, can you do more research on this? The string might even be different depending on the phase, if so, that's fine, we just need to define the strings separately.